### PR TITLE
MAG-354: Allow success order placement on oney pending payment

### PR DESCRIPTION
**Pull request by TristanBouchierAgenceDnD**
changelog: fixed

# ⚠️ Requirements
Reviewer, please take a look at those requirements before merging on `master` :

- [X] Check that plugin version has been upgaded and are identical in both `composer.json` and `Helper/Config.php` files

### CODE REVIEW

File: Controller/Payment/PaymentReturn.php [hunk 1]
```diff
@@ -113,4 +87,23 @@ public function execute()
             return $resultRedirect->setPath($redirectUrlSuccess);
         }
     }
+
+    /**
+     * Return true if we are paying with oney and the payment isn't rejected but waiting for approval
+     *
+     * @param Payment|null $payment
+     *
+     * @return bool
```
Comment (AntoineKociubaAgenceDnD) [inline Controller/Payment/PaymentReturn.php:96]: to remove, useless

File: Controller/Payment/PaymentReturn.php [hunk 2]
```diff
@@ -113,4 +87,23 @@ public function execute()
             return $resultRedirect->setPath($redirectUrlSuccess);
         }
     }
+
+    /**
+     * Return true if we are paying with oney and the payment isn't rejected but waiting for approval
+     *
+     * @param Payment|null $payment
```
Comment (AntoineKociubaAgenceDnD) [inline Controller/Payment/PaymentReturn.php:94]: to remove, useless

File: Controller/Payment/PaymentReturn.php [hunk 3]
```diff
@@ -113,4 +87,23 @@ public function execute()
...
+     *
+     * @return bool
+     */
+    public function isOneyPending(?Payment $payment): bool
+    {
+        if ($payment) {
+            $paymentMethod = $payment->payment_method;
+            if ($payment->is_paid === false && $paymentMethod['is_pending'] && $paymentMethod['type']) {
+                return (str_contains($paymentMethod['type'], 'oney') && $paymentMethod['is_pending'] === true);
```
Comment (AntoineKociubaAgenceDnD) [inline Controller/Payment/PaymentReturn.php:103]: Which PHP version should we remain compatible with ? ⚠️

File: Controller/Payment/PaymentReturn.php [hunk 4]
```diff
@@ -1,48 +1,23 @@
...
-        $this->orderRepository = $orderRepository;
-    }
-
     /**
      * Handle return from PayPlug payment page
      *
      * @return mixed
      */
-    public function execute()
+    public function execute(): mixed
```
Comment (AntoineKociubaAgenceDnD) [inline Controller/Payment/PaymentReturn.php:20]: `mixed` is not a proper type. Type properly and more specifically instead, keeping in mind PHP compatibility constraints also

File: Controller/Payment/PaymentReturn.php [hunk 5]
```diff
@@ -1,48 +1,23 @@
...
-        OrderRepository $orderRepository
-    ) {
-        parent::__construct($context, $checkoutSession, $salesOrderFactory, $logger, $payplugHelper);
-        $this->orderRepository = $orderRepository;
-    }
-
     /**
      * Handle return from PayPlug payment page
      *
      * @return mixed
```
Comment (AntoineKociubaAgenceDnD) [inline Controller/Payment/PaymentReturn.php:18]: to remove, useless

**Review by AntoineKociubaAgenceDnD — CHANGES_REQUESTED** [review]

File: Controller/Payment/PaymentReturn.php (hunk 5 above)
Comment (TristanBouchierAgenceDnD) [reply Controller/Payment/PaymentReturn.php:18]: OK

File: Controller/Payment/PaymentReturn.php (hunk 3 above)
Comment (TristanBouchierAgenceDnD) [reply Controller/Payment/PaymentReturn.php:103]: For now 8.0+, Antoine will see with Dani, but all the developments have been made so far in this direction

File: Controller/Payment/PaymentReturn.php (hunk 2 above)
Comment (TristanBouchierAgenceDnD) [reply Controller/Payment/PaymentReturn.php:94]: OK

File: Controller/Payment/PaymentReturn.php (hunk 1 above)
Comment (TristanBouchierAgenceDnD) [reply Controller/Payment/PaymentReturn.php:96]: OK

File: Controller/Payment/PaymentReturn.php (hunk 4 above)
Comment (TristanBouchierAgenceDnD) [reply Controller/Payment/PaymentReturn.php]: OK

**Review by TristanBouchierAgenceDnD — COMMENTED** [review]
Changes were made

**Review by AntoineKociubaAgenceDnD — APPROVED** [review]

**Pull request merged**